### PR TITLE
refactor: remove sway wording

### PR DESCRIPTION
**Pull request by hal3e**
closes:https://github.com/FuelLabs/fuels-rs/issues/257

- AFAIK the rust SDK does not depend on any Sway functionality.
- I have tried to remove all the Sway mentions from the code and left the documentation intact. I need the teams opinion on the wording I used :)

### CODE REVIEW

File: packages/fuels-core/src/abi_encoder.rs [hunk 1]
```diff
@@ -104,7 +104,7 @@ impl ABIEncoder {
 
         self.encode_discriminant(*discriminant);
 
-        // The sway compiler has an optimization for enums which have only Units as variants -- such
+        // The high-level language compiler has an optimization for enums which have only Units as variants -- such
```
Comment (MujkicA) [inline packages/fuels-core/src/abi_encoder.rs:107]: Consistency

File: packages/fuels-core/src/abi_decoder.rs [hunk 2]
```diff
@@ -216,7 +216,7 @@ impl ABIDecoder {
         selected_variant: &ParamType,
         enum_width: usize,
     ) -> Result<Token, CodecError> {
-        // The sway compiler has an optimization where enums that only contain
+        // The FuelVM has an optimization where enums that only contain
```
Comment (MujkicA) [inline packages/fuels-core/src/abi_decoder.rs:219]: Nit: be consistent about the wording here

**Review by MujkicA — APPROVED** [review]

**Review by Salka1988 — APPROVED** [review]
Looks good! You've got them all.

File: packages/fuels-core/src/abi_decoder.rs (hunk 2 above)
Comment (hal3e) [reply packages/fuels-core/src/abi_decoder.rs:219]: Do you guys prefer `FuelVM` or `high-level language` ?

File: packages/fuels-core/src/abi_encoder.rs (hunk 1 above)
Comment (hal3e) [reply packages/fuels-core/src/abi_encoder.rs:107]: Good catch! That was me changing stuff and not updating everywhere...

File: packages/fuels-core/src/abi_decoder.rs (hunk 2 above)
Comment (segfault-magnet) [reply packages/fuels-core/src/abi_decoder.rs:219]: What bothers me is that, as far as I understand it, the VM could care less about enums. 

It was the sway compiler that had this optimization. But how do you say that without saying sway :D 

High-level-language compiler has an optimization?

Or just be vague, my favourite, "Enums that contain only units are bla bla" don't say why :)

**Review by segfault-magnet — COMMENTED** [review]
Soo, I have a good feeling the following is overkill, but just want to get it out there -- 

What do you think about a CI step which does a search of our docs and `.rs` files for sway wording? 

Or we will be careful to not introduce it back :D

**Comment by hal3e** [comment]
> Soo, I have a good feeling the following is overkill, but just want to get it out there --
> 
> What do you think about a CI step which does a search of our docs and `.rs` files for sway wording?
> 
> Or we will be careful to not introduce it back :D

That is definitely an overkill :)

File: packages/fuels-core/src/abi_decoder.rs (hunk 2 above)
Comment (digorithm) [reply packages/fuels-core/src/abi_decoder.rs:219]: That's a good point. In this case, this optimization doesn't happen in the VM. 

I'm ok with what @segfault-magnet said, don't go too deep into why this optimization is done.

File: packages/fuels-core/src/code_gen.rs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 //! This module implements everything related to code generation/expansion
-//! from a fuel/sway ABI.
+//! from a fuel ABI.
```
Comment (digorithm) [inline packages/fuels-core/src/code_gen.rs:2]:
```suggestion
//! from a FuelVM ABI.
```
For consistency!

File: tools/fuels-abi-cli/src/main.rs [hunk 4]
```diff
@@ -5,7 +5,7 @@
 // TODO: currently disabled because it needs to be updated to the new JSON ABI.
 
 // #[derive(StructOpt, Debug)]
-// /// Sway/Fuel ABI coder.
+// /// Fuel ABI coder.
```
Comment (digorithm) [inline tools/fuels-abi-cli/src/main.rs:8]:
```suggestion
// /// FuelVM ABI coder.
```

**Review by digorithm — CHANGES_REQUESTED** [review]
Very good! Just some tiny nits.

**Review by iqdecay — DISMISSED** [review]

**Review by iqdecay — APPROVED** [review]

**Review by digorithm — APPROVED** [review]

**Pull request merged**